docs/upgrade-govuk-frontend.md

```diff
@@ -0,0 +1,24 @@
+# Keep this gem in sync with the Design System
+
+This gem consumes [`govuk-frontend`](https://github.com/alphagov/govuk-frontend) via [Yarn](https://classic.yarnpkg.com/). [Dependabot](https://github.com/marketplace/dependabot-preview) will let us know when [a new version of `govuk-frontend` is available](https://github.com/alphagov/govuk_publishing_components/pulls?q=is%3Apr+is%3Aopen+label%3Adependencies+bump+govuk-frontend). There are a few checks to be done before merging in a 'Bump `govuk-frontend`' pull request.
+
+1. Read the [release notes](https://github.com/alphagov/govuk-frontend/releases/). GOV.UK Frontend has well-documented release notes that, depending on the release type, might contain the following sections:
+ - Breaking changes (major release)
+ - New features (minor release)
+ - Fixes (patch release)
+
+2. Check if and how the breaking changes affect this gem and GOV.UK applications dependent on this gem. For each breaking change follow the migration steps in the release notes and push the changes in a separate commit on the same Dependabot branch. Make sure to test GOV.UK applications with this branch. On major releases is always a good idea to involve someone from [the Design System team](https://github.com/orgs/alphagov/teams/team-gov-uk-design-system/members) in the review process as they have a thorough understanding of the changes.
```

Comment:
I see what you've done here with using points 2, 3 and 4 to reflect breaking changes, new features and fixes respectively, however it took me a moment to realise that's what the intent was and I got a bit confused going from 2 to 3. I wonder if these paragraphs would benefit being under a single list item and separated by headings ("For any breaking changes in this release..."). I appreciate that that's going to break markdown formatting and might generally be a faff to make look nice so I'm not precious about this.

Reply:
I see your point, Owen. I have restructured the documentation into 3 main steps in the last commit. Let me know if this addresses your point.

Reply:
Exceptional. Cheers Alex.